File: Q1_TrianglePattern.py
# right half pyramid
def right_half_triangle(rows):
    for i in range (1,rows+1):
        stars = '*' * i
        print(stars)

right_half_triangle(5)


# left half pyramid
def left_half_triangle(rows):
    for i in range (1, rows + 1):
        spaces = ' ' * (rows - i)
        stars = '*' * i
        print(spaces + stars)

left_half_triangle(5)

# full pyramid
def full_pyramid(rows):
    for i in range (1, rows + 1):
        spaces = ' ' * (rows - i)
        stars = '*' * (i*2 - 1)
        print (spaces + stars)

full_pyramid(5)

File: Q1_TrianglePattern.js
// right half triangle
function right_half_triangle(rows){
    for (let i=1; i <= rows; i++){
        let stars = '*'.repeat(i);
        console.log(stars);
    }
}

right_half_triangle(5);

// left half triangle
function left_half_triangle(rows){
    for (let i=1; i <= rows; i++){
        let spaces = ' '.repeat(rows - i);
        let stars = '*'.repeat(i);
        console.log(spaces + stars)
    }
}

left_half_triangle(5);

// full triangle
function full_triangle(rows){
    for (let i=1; i<=rows; i++){
        let spaces = ' '.repeat(rows-i)
        let stars = '*'.repeat(i*2 - 1)
        console.log(spaces + stars)
    }
}

full_triangle(5);